### PR TITLE
CURATOR-585: fix DiscoveryExample did not check for exceptions

### PR DESCRIPTION
**Pull request by iwangjie**

### CODE REVIEW

File: curator-examples/src/main/java/discovery/DiscoveryExample.java [hunk 1]
```diff
@@ -192,8 +194,13 @@ private static void listInstances(ServiceDiscovery<InstanceDetails> serviceDisco
                 }
             }
         }
-        finally
```
Comment (eolivelli) [inline curator-examples/src/main/java/discovery/DiscoveryExample.java:195]: Please keep the closing of resources into the finally block

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: curator-examples/src/main/java/discovery/DiscoveryExample.java (hunk 1 above)
Comment (iwangjie) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java:195]: OK, thank you. I'll put the resource shutdown in the finally block.

File: curator-examples/src/main/java/discovery/DiscoveryExample.java [hunk 2]
```diff
@@ -192,6 +194,14 @@ private static void listInstances(ServiceDiscovery<InstanceDetails> serviceDisco
                 }
             }
         }
+        catch ( Exception e )
+        {
+            if ( e instanceof KeeperException.NoNodeException )
```
Comment (eolivelli) [inline curator-examples/src/main/java/discovery/DiscoveryExample.java:199]: which instruction can throw this exception ? is it possible that it is throws inside the inner loop ? 
in that case we could end the loop prematurely .
that do you think ?

Comment (iwangjie) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java:199]: When there is no one service in the registration call list will directly exit the process, and in fact should be prompted without any services, unless it is other exceptions, do you think?

Comment (iwangjie) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java:199]: Maybe, if it's not a NoNodeException, I should throw it to stop the execution of the program, right?

Comment (eolivelli) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java:199]: let's make it simple.
if you found this problem with NoNode then let's fix only it.

if you have ever saw other errors, then it is worth to do more

what do you think ?

File: curator-examples/src/main/java/discovery/DiscoveryExample.java [hunk 3]
```diff
@@ -192,6 +194,14 @@ private static void listInstances(ServiceDiscovery<InstanceDetails> serviceDisco
...
+            {
+                System.err.println("There are no registered instances.");
+                return;
+            }
```
Comment (eolivelli) [inline curator-examples/src/main/java/discovery/DiscoveryExample.java:203]: any other exception should be rethrown

Comment (iwangjie) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java:203]: Yes, that's what I thought, and I hope what I'm doing now is the same as what we discussed.

File: curator-examples/src/main/java/discovery/DiscoveryExample.java (hunk 2 above)
Comment (iwangjie) [reply curator-examples/src/main/java/discovery/DiscoveryExample.java]: OK, it has been modified

**Comment by iwangjie** [comment]
Do I have anything to deal with?

**Review by eolivelli — APPROVED** [review]
sorry for late reply.
I am merging this patch now

**Comment by eolivelli** [comment]
committed to master branch
thanks

**Pull request closed without merging**